src/components/admin/MiscConfig.js

```diff
@@ -0,0 +1,113 @@
+import React, {Component} from 'react';
+import {
+    Badge,
+    Button,
+    Input,
+    InputGroup,
+    InputGroupAddon,
+    InputGroupText
+} from "reactstrap";
+
+import DB from "../../state/DB";
+
+export default class MiscConfig extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            old_password: "",
+            new_password: "",
+            confirm_password: "",
+            show_password_input: false,
+        };
+
+        this.toggleSignUps = this.toggleSignUps.bind(this);
+        this.togglePasswordInput = this.togglePasswordInput.bind(this);
+        this.handlePasswordChange = this.handlePasswordChange.bind(this);
+        this.handlePasswordSubmit = this.handlePasswordSubmit.bind(this);
+    }
+
+    toggleSignUps() {
+        DB.setConfig({...DB.config, ...{"sign_ups": !DB.config.sign_ups}});
+        DB.updateConfigFile();
+        this.props.refresh();
+    }
+
+    togglePasswordInput() {
+        this.setState({
+            show_password_input: !this.state.show_password_input,
+        })
+    }
+
+    handlePasswordChange(event, inputName) {
```

Comment:
why don't you just make the name in state the actual name attribute of the element so you can use `event.target.name` and not complicate this function